File: OJ/剑指Offer/二维数组中的查找.py
# 思路：
# 对于左下角的元素，向右递增，向上递减
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        width = len(array[0])
        height = len(array)
        i, j = height-1, 0
        while i>=0 and j < width:
            if array[i][j] == target:
                return True
            if array[i][j] > target:
                i -= 1
            elif array[i][j] < target:
                j += 1
        return False

File: OJ/剑指Offer/二维数组中的查找.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        // 思路： 从左下角开始，大了就向上，小了就向右

        int row=0;
            int col=array[0].length-1;
            while(row<=array.length-1&&col>=0){
                if(target==array[row][col])
                    return true;
                else if(target>array[row][col])
                    row++;
                else
                    col--;
            }
            return false;
    }
}